src/Graphics/Image/Processing/Otsu.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE BangPatterns #-} 
+
+module Graphics.Image.Processing.Otsu where
+
+import Control.Monad as CM (forM_, when)
+import Control.Monad.ST
+import Data.STRef 
+
+import Prelude as P hiding (foldl', foldr)
+import Graphics.Image.Processing.Filter
+import Graphics.Image.Interface as I
+import Graphics.Image
+import Graphics.Image.Types as IP
+import Graphics.Image.ColorSpace (X)
+import Data.Vector.Unboxed as V
+import Data.List as L
+
+-- the tail is safe because scanl alawys returns a nonempty list
```

Comment:
That statement is incorrect. `dropWhile` could make that list empty, so it is important to know that there is at least one element in the list with `wF == 0`. For example, this function will produce an error if `hist` is empty.